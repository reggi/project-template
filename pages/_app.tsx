import * as React from "react";

import { DefaultTheme, Provider as PaperProvider } from "react-native-paper";

import { AppProps } from "next/app";
import Head from "next/head";

const theme = {
  ...DefaultTheme,
};

export default function MyApp(props: AppProps) {
  const { Component, pageProps } = props;
  return (
    <>
      <Head>
        <title>Change title in _app.tsx</title>
        <meta name="viewport" content="initial-scale=1, width=device-width" />
      </Head>
      <PaperProvider theme={theme}>
        <Component {...pageProps} />
      </PaperProvider>
    </>
  );
}
