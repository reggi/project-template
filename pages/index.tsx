import React, { FC, useEffect, useState } from "react";
import { StyleSheet, Text, View } from "react-native";

import { fetch } from "whatwg-fetch";

function useRandomNumber() {
  const [number, setNumber] = useState<number>();

  useEffect(() => {
    fetch("http://localhost:3000/api/randomNumber")
      .then(async (response) => await response.text())
      .then((text) => setNumber(+text));
  }, []);

  return number;
}

export const Index: FC = () => {
  const number = useRandomNumber();
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Welcome to Expo + Next.js</Text>
      <Text>Random Number: {number}</Text>
    </View>
  );
};

export default Index;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontSize: 16,
  },
});
