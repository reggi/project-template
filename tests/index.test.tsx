import {Index} from '../pages/index';
import React from 'react'
import renderer from 'react-test-renderer';

test('should render index page', () => {
    const index = renderer.create(<Index />)
    expect(index.toJSON()).toMatchSnapshot();
});