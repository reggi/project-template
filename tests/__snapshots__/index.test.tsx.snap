// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render index page 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    Welcome to Expo + Next.js
  </Text>
  <Text>
    Random Number: 
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 12,
              "marginRight": -4,
            }
          }
        >
          <Text />
        </View>
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "papayawhip",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "palevioletred",
          },
        ]
      }
    >
      Hello World!
    </Text>
  </View>
</View>
`;
